envoy/api/v2/core/base.proto

```diff
@@ -39,6 +39,19 @@ message Percent {
   double value = 1 [(validate.rules).double = {gte: 0, lte: 100}];
 }
 
+// A truncated percentage is used in cases in which for performance reasons performing floating
+// point to integer conversions during randomness calculations is undesirable. The message includes
+// both a numerator and denominator that together determine the final truncated percentage in the
+// range 0-100 inclusive.
+message TruncatedPercent {
```

Comment:
How come not in `envoy.types`?

Reply:
Because this is where the existing percentage type is. Do you want me to move them both over to the types package? Don't care either way.

Reply:
I think maybe it makes sense. @wora?